examples/stdlib/list/countIf.effekt

```diff
@@ -0,0 +1,17 @@
+module examples/pos/list/countIf
+
+import list
+
+def main() = {
+  val empty: List[String] = []
+  println(empty.countIf{ x => x.length() == 2}) // expects 0
```

Comment:
Style (applies for all of the examples both here and in `list`)
- space before block arguments
- spacing inside block arguments
```suggestion
  val empty: List[String] = []
  println(empty.countIf { x => x.length == 2 }) // expects 0
```